primitives/tracing/src/proxied_span.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Comprises the struct `ProxiedTrace` required by the proxy, encapsulates
+//! the necessary associated unsafe code and provides a safe interface.
+
+use std::pin::Pin;
+use tracing::{span::Entered, Span};
+
+/// Container for a proxied tracing Span and it's associated guard
+pub struct ProxiedSpan {
+	id: u64,
+	// Note that `guard` and `span` must be declared in this order so that `guard` is dropped first.
```

Comment:
Please write a custom `Drop` impl so ensure this!

Reply:
Struct field drop order is now stable <https://doc.rust-lang.org/reference/destructors.html>, <https://github.com/rust-lang/rfcs/pull/1857> but agree it's good to be explicit about this. In terms of explicitness it seems like `ManuallyDrop` is a better choice than `Option::take()`, even though it requires another block of unsafe code.

Reply:
Yeah I know that this is defined, but better safe than sorry :D